### PR TITLE
[IIIF-1054] Support multiple media mount points

### PR DESCRIPTION
**Pull request by ksclarke**
* Support multiple media mount points [IIIF-1054]
* Improve exception handling and logging [IIIF-1052]
* Change static methods to regular instance methods
* Use try-with-resources syntax for CSVReader and CSVWriter

### CODE REVIEW

File: src/main/resources/metagetter_messages.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  <entry key="MG-100">File path must exist: {}</entry>
+  <entry key="MG-101">Problem reading file or walking the supplied directory for files: {}</entry>
+  <entry key="MG-102">The ffprobe application cannot be found on the supplied path: {}</entry>
+  <entry key="MG-103">Adding A/V metadata to supplied CSV file: {}</entry>
+  <entry key="MG-104">Problem reading/writing CSV file: {}</entry>
+  <entry key="MG-105">File '{}' not found in following paths: {}</entry>
+  <entry key="MG-106">Problem reading media file: {}</entry>
+  <entry key="MG-107">Supplied CSV file does not contain the required 'File Name' column</entry>
+  <entry key="MG-108">File does not have the expected extension indicating media type: {}</entry>
+</properties>
```
Comment (DRickard) [inline src/main/resources/metagetter_messages.xml]: FYI: GitHub is complaining about a missing newline at EOF

File: src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java [hunk 2]
```diff
@@ -50,47 +62,66 @@
...
     /**
-     * Return code for non-existent file/directory.
+     * Cleans up after a MetadataSetter test.
+     *
+     * @throws IOException If there is a problem deleting the file
      */
-    private static final int NO_FILE = 101;
+    @After
+    public void tearDown() throws IOException, SecurityException {
+        // Files.deleteIfExists(FileSystems.getDefault().getPath(OUTPUT_PATH + CSV_NAME));
```
Comment (DRickard) [inline src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java]: Just curious: why keep the `tearDown()` method when the only line in it is commented out?

**Review by DRickard — DISMISSED** [review]

File: src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java (hunk 2 above)
Comment (ksclarke) [reply src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java]: Oops, thanks. That's a mistake. It should still be in there. I just commented out to test something. I'll uncomment.

File: src/main/resources/metagetter_messages.xml (hunk 1 above)
Comment (ksclarke) [reply src/main/resources/metagetter_messages.xml]: Thanks. Fixed. Not sure why Checkstyle didn't catch that. I'll have to look at the Checkstyle rules.

**Review by DRickard — DISMISSED** [review]

**Review by DRickard — DISMISSED** [review]

File: src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java [hunk 3]
```diff
@@ -32,7 +39,12 @@
     /**
      * Path to media files to be read.
      */
-    private static final String MEDIA_PATH = "src/test/resources/media/";
+    private static final String MEDIA_PATH = TEST_FIXTURES_DIR + "media/";
+
+    /**
+     * A secondary path from which media files can be read.
```
Comment (markmatney) [inline src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java]: I think this comment should explain that it's a comma-delimited string of paths (maybe also have another constant for the second path)

**Review by markmatney — CHANGES_REQUESTED** [review]

File: src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java (hunk 3 above)
Comment (ksclarke) [reply src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java]: You mean the comment for line 47? A little confused by the inclusion of line 42.

Comment (markmatney) [reply src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java]: Yeah, the comment for line 47. I think the GitHub UI is just showing line 42-44 for context (I had only selected line 45 when I made this comment).

Comment (ksclarke) [reply src/test/java/edu/ucla/library/services/metadata/MetadataSetterTest.java]: Comment updated to make it clearer that the path should be a comma delimited string.

**Review by markmatney — APPROVED** [review]

**Review by DRickard — APPROVED** [review]

**Pull request merged**